packages/date-picker/src/vaadin-date-picker-overlay-content-mixin.js

```diff
@@ -361,10 +374,14 @@ export const DatePickerOverlayContentMixin = (superClass) =>
      * @protected
      */
     _selectDate(dateToSelect) {
+      if (!this._dateAllowed(dateToSelect)) {
```

Comment:
Might need to discuss this change. I updated this to be a boolean function so that the keyboard `Enter` behavior could be prevented for disabled dates. That allows us to hover on disabled dates without allowing selection of those dates.